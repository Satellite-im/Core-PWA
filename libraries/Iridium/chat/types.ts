export type ConversationMessage = {
  conversationId: string
  from: string
  body: string
  at: number
  type: 'text' | 'file' | 'group' | 'glyph' | 'divider' | 'string'
  attachments: string[]
}

export type Conversation = {
  id: string
  type: 'direct' | 'group'
  name?: string
  participants: string[]
  createdAt: number
  updatedAt: number
  messages: {
    [key: string]: ConversationMessage
  }
  messageArray: ConversationMessage[]
}

export const ChatError = {
  CONVERSATION_EXISTS: 'error.chat.conversation_exists',
  CONVERSATION_NOT_FOUND: 'error.chat.conversation_not_found',
  MESSAGE_NOT_SENT: 'error.chat.message_not_sent',
}
