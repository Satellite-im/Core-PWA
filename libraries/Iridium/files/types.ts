/* eslint-disable no-use-before-define */
import { DIRECTORY_TYPE } from '~/libraries/Files/types/directory'
import { FILE_TYPE } from '~/libraries/Files/types/file'

export type FileUpdate = 'upload' | 'delete' | 'like' | 'share'

interface Shared {
  id: string
  name: string
  liked: boolean
  shared: boolean
  modified: number
  size: number
  parentId: IridiumDirectory['id'] // empty string if root item
}

export interface IridiumFile extends Shared {
  type: FILE_TYPE
  description?: string
  thumbnail: string
  extension: string
  nsfw: boolean
}

export interface IridiumDirectory extends Shared {
  type: DIRECTORY_TYPE
  children: IridiumItem[]
}

export type IridiumItem = IridiumFile | IridiumDirectory
