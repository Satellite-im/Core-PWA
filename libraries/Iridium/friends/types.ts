import { User } from '../users/types'

export type Friend = User & {}

export type FriendRequestStatus =
  | 'pending'
  | 'accepted'
  | 'rejected'
  | 'removed'

export type FriendRequest = {
  did: User['did']
  incoming: boolean
  status: FriendRequestStatus
  at: number
}

export const FriendsError = {
  NETWORK_ERROR: 'errors.friends.network',
  FRIEND_EXISTS: 'errors.friends.exists',
  FRIEND_NOT_FOUND: 'errors.friends.not_found',
  REQUEST_NOT_FOUND: 'errors.friends.request_not_found',
  REQUEST_NOT_SENT: 'errors.friends.request_not_sent',
  REQUEST_ALREADY_SENT: 'errors.friends.request_already_sent',
}
