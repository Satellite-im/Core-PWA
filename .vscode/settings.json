{
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "editor.formatOnSave": true,
  "[vue]": {
    "editor.defaultFormatter": "esbenp.prettier-vscode"
  },
  "cSpell.words": [
    "Dexie",
    "minisearch"
  ],
  "nuxt.isNuxtApp": true
}
