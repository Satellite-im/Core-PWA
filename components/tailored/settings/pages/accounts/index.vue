<template src="./Accounts.html"></template>

<script lang="ts">
import Vue from 'vue'

export default Vue.extend({
  name: 'AccountsSettings',
  layout: 'settings',
})
</script>
