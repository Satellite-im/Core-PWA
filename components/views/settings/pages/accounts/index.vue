<template src="./Accounts.html"></template>

<script lang="ts">
import Vue from 'vue'
import { mapState } from 'vuex'

import { ClipboardIcon } from 'satellite-lucide-icons'
import { RootState } from '~/types/store/store'
import iridium from '~/libraries/Iridium/IridiumManager'

export default Vue.extend({
  name: 'AccountsSettings',
  components: {
    ClipboardIcon,
  },
  layout: 'settings',
  data() {
    return {
      showPhrase: false,
      profile: iridium.profile.state,
    }
  },
  computed: {
    ...mapState({
      accounts: (state) => (state as RootState).accounts,
    }),
    getId(): string | unknown {
      if (!iridium.connector) return
      return this.profile
        ? `${this.profile.name}#${iridium.id.substring(iridium.id.length - 6)}`
        : `${iridium.id}`
    },
    storePin: {
      set(state) {
        this.$store.commit('accounts/setStorePin', state)
      },
      get(): boolean {
        return this.accounts.storePin
      },
    },
    splitPhrase(): string[] {
      return this.accounts.phrase.split(' ')
    },
  },
  methods: {
    /**
     * @method togglePhrase DocsTODO
     * @description
     * @example
     */
    togglePhrase() {
      this.showPhrase = !this.showPhrase
    },
    copyAddress(copyText: string) {
      navigator.clipboard.writeText(copyText)
      this.$toast.show(this.$t('ui.copied') as string)
    },
    copyPhrase() {
      navigator.clipboard.writeText(this.accounts.phrase)
      this.$toast.show(this.$t('ui.copied') as string)
    },
  },
})
</script>

<style scoped lang="less" src="./Accounts.less"></style>
