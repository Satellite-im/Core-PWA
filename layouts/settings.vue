<template src="./settings/Settings.html"></template>

<script lang="ts">
import Vue from 'vue'
import { mobileSwipe } from '../components/mixins/Swipe/Swipe'
export default Vue.extend({
  mixins: [mobileSwipe],
  middleware: 'authenticated',
  data() {
    return {
      sidebar: true,
    }
  },
  watch: {
    // We want to close the sidebar when on a mobile device if a user
    // selects a setting group. This watches route changes and toggles
    // the sidebar if it's on a mobile device.
    $route() {
      // @ts-ignore
      if (this.$route.name?.includes('settings') && this.$device.isMobile) {
        this.sidebar = false
      }
    },
  },
  methods: {
    /**
     * Opens and closes the left hand sidebar
     */
    toggleSidebar() {
      this.$data.sidebar = !this.$data.sidebar
    },
  },
})
</script>

<style lang="less" src="./settings/Settings.less"></style>
