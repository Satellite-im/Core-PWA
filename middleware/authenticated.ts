import { NuxtRouteConfig } from '@nuxt/types/config/router'
import memoize from 'lodash/memoize'
// TODO: verify why we got the import/named error for RawLocation import AP-394
// eslint-disable-next-line import/named
import { RawLocation } from 'vue-router'
import { Config } from '~/config'
import { RootStore } from '~/types/store/store'
import iridium from '~/libraries/Iridium/IridiumManager'
interface Arguments {
  store: RootStore
  redirect: (location: RawLocation) => void
  route: NuxtRouteConfig
}

/**
 * @method
 * @description
 * @param
 * @returns
 * @example
 */
export default function ({ store, route, redirect }: Arguments) {
  const { locked, phrase } = store.state.accounts

  const eventuallyRedirect = memoize(
    (path: string) => {
      if (route.path === path) return
      redirect(path)
    },
    () => redirect,
  )

  // If the user is not authenticated
  if (locked) {
    return eventuallyRedirect('/auth/unlock')
  }

  // If the wallet has not been created yet
  if (!locked && phrase === '' && !route.path.includes('setup')) {
    return eventuallyRedirect('/setup/disclaimer')
  }

  const [_, domain] = route.path.split('/')

  if (Config.routingMiddleware.prerequisitesCheckBypass.includes(domain)) {
    return
  }

  if (!iridium.ready) {
    return eventuallyRedirect('/')
  }

  if (route && route.path === '/') {
    return eventuallyRedirect('/friends')
  }

  store.commit('accounts/setLastVisited', route.path)
}
