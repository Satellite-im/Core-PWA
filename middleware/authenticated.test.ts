import * as authenticated from './authenticated'
import { RegistrationStatus } from '~/store/accounts/types'
import { DataStateType } from '~/store/dataState/types'
import { CaptureMouseTypes } from '~/store/settings/types'
import { RootState } from '~/types/store/store'

const initialRootState: RootState = {
  accounts: {
    storePin: true,
    loading: true,
    locked: true,
    pin: '',
    pinHash: '',
    active: 'true',
    gasPrice: '',
    phrase: '',
    error: '',
    encryptedPhrase: '',
    registered: true,
    details: {
      name: '',
      address: '',
      status: '',
      state: 'idle',
      unreadCount: 123,
      profilePicture: '',
      badge: 'community',
      userAccount: '',
      mailboxId: '',
      textilePubkey: '',
    },
    registrationStatus: RegistrationStatus.IN_PROGRESS,
    lastVisited: '',
  },
  dataState: {
    files: DataStateType.Empty,
    friends: DataStateType.Loading,
    search: DataStateType.Ready,
  },
  friends: {
    incomingRequests: [
      {
        requestId: '',
        account: {
          accountId: '',
          from: '',
          status: 123,
          fromMailboxId: '',
          toMailboxId: '',
          to: '',
        },
        pending: true,
        from: '',
        userInfo: {
          name: '',
          servers: {},
          status: '',
          photoHash: '',
        },
      },
    ],
    outgoingRequests: [
      {
        to: '',
        requestId: '',
        account: {
          accountId: '',
          from: '',
          status: 123,
          fromMailboxId: '',
          toMailboxId: '',
          to: '',
        },
        pending: true,
      },
    ],
    all: [
      {
        publicKey: 'NoWiFi4you',
        typingState: 'NOT_TYPING',
        item: {},
        pending: true,
        activeChat: true,
        encryptedTextilePubkey: '',
        name: 'Taurus Nix',
        address: '0xdf9eb223bafbe5c5271415c75aecd68c21fe3d7f',
        account: {
          accountId: 'Checking Account',
          from: '.',
          status: 429,
          fromMailboxId: '12345',
          toMailboxId: 'v4.0.0-rc.4',
          to: './path/to/file',
        },
        textilePubkey: 'https://accounts.google.com/o/oauth2/revoke?token=%s',
        status: '',
        state: 'idle',
        unreadCount: 123,
        profilePicture: '',
        badge: 'community',
        userAccount: '',
        mailboxId: '',
      },
    ],
  },
  textile: {
    initialized: true,
    conversations: {},
    conversationLoading: true,
    messageLoading: true,
    uploadProgress: {
      abc: {
        progress: 42,
        finished: false,
        name: 'file.pdf',
      },
    },
  },
  webrtc: {
    initialized: true,
    incomingCall: '',
    activeCall: '',
    connectedPeers: [],
    activeStream: {
      createdAt: 0,
    },
    streaming: true,
    localTracks: { audio: {}, video: {} },
    remoteTracks: { audio: {}, video: {} },
  },
  settings: {
    audioInput: '',
    audioOutput: '',
    videoInput: '',
    captureMouse: CaptureMouseTypes.always,
    noiseSuppression: true,
    echoCancellation: true,
    bitrate: 96000,
    sampleSize: 24,
    timezone: Intl.DateTimeFormat().resolvedOptions().timeZone,
    userHasGivenAudioAccess: false,
    userDeniedAudioAccess: false,
    keybinds: {
      toggleMute: 'alt+m',
      toggleDeafen: 'alt+d',
      openSettings: 'alt+s',
      callActiveChat: 'alt+c',
    },
    embeddedLinks: true,
    displayCurrentActivity: true,
  },
}

describe('', () => {
  test('the user is not authenticated', () => {
    const inst = authenticated.default
    const param1 = {
      state: initialRootState,
      getters: { allPrerequisitesReady: true },
    }
    const param2: () => void = () => {}
    const param3 = { chunkName: 'string', path: 'path' }

    expect(
      inst({
        store: param1,
        redirect: param2,
        route: param3,
      }),
    ).toMatchSnapshot()
  })
  test('the wallet has not been created yet', () => {
    const inst = authenticated.default
    initialRootState.accounts.locked = false
    initialRootState.accounts.phrase = ''
    const param1 = {
      state: initialRootState,
      getters: { allPrerequisitesReady: true },
    }
    const param2: () => void = () => {}
    const param3 = { chunkName: 'string', path: 'https://example.com/page1' }

    expect(
      inst({
        store: param1,
        redirect: param2,
        route: param3,
      }),
    ).toMatchSnapshot()
  })
  test('param3 path included in config', () => {
    const inst = authenticated.default
    initialRootState.accounts.locked = false
    initialRootState.accounts.phrase = ''
    const param1 = {
      state: initialRootState,
      getters: { allPrerequisitesReady: true },
    }
    const param2: () => void = () => {
      return 'https://example.com/page1'
    }
    const param3 = { chunkName: 'string', path: 'ab/setup' }

    expect(
      inst({
        store: param1,
        redirect: param2,
        route: param3,
      }),
    ).toMatchSnapshot()
  })
  test('all prerequisites not ready', () => {
    const inst = authenticated.default
    initialRootState.accounts.locked = false
    initialRootState.accounts.phrase = ''
    const param1 = {
      state: initialRootState,
      getters: { allPrerequisitesReady: false },
    }
    const param2: () => void = () => {
      return 'https://example.com/page1'
    }
    const param3 = { chunkName: 'string', path: 'ab/setup' }

    expect(
      inst({
        store: param1,
        redirect: param2,
        route: param3,
      }),
    ).toMatchSnapshot()
  })
  test('store.commit called', () => {
    const inst = authenticated.default
    initialRootState.accounts.locked = false
    initialRootState.accounts.phrase = ''
    const param1 = {
      state: initialRootState,
      getters: { allPrerequisitesReady: true },
      commit: jest.fn(),
    }
    const param2: () => void = () => {
      return 'https://example.com/page1'
    }
    const param3 = { chunkName: 'string', path: 'ab/notsetup' }

    expect(
      inst({
        store: param1,
        redirect: param2,
        route: param3,
      }),
    ).toMatchSnapshot()
    expect(param1.commit).toHaveBeenCalled()
  })
})
