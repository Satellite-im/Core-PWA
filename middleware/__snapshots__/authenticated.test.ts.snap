// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` all prerequisites not ready 1`] = `undefined`;

exports[` param3 path included in config 1`] = `undefined`;

exports[` store.commit called 1`] = `undefined`;

exports[` store.commit called 2`] = `undefined`;

exports[` the user is not authenticated 1`] = `undefined`;

exports[` the wallet has not been created yet 1`] = `undefined`;
